### PR TITLE
Add RuleDetails component

### PR DESCRIPTION
**Pull request by bmclaughlin**
Moving linking the list of tags to a separate PR, PR link coming soon.

rule-details

### CODE REVIEW

File: src/AppConstants.js [hunk 1]
```diff
@@ -59,3 +59,18 @@ export const FILTER_CATEGORIES = [
         ]
     }
 ];
+export const RISK_OF_CHANGE_LABEL = [ 'Very Low', 'Low', 'Moderate', 'High' ];
+export const RISK_OF_CHANGE_DESC = [
```
Comment (AllenBW) [inline src/AppConstants.js]: Kinda feels like this should be an object, where the key is the label, and the description is the value

File: src/PresentationalComponents/RuleDetails/RuleDetails.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
+import React, { Component } from 'react';
+import { Ansible, Battery, Shield, routerParams } from '@red-hat-insights/insights-frontend-components';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Grid, GridItem } from '@patternfly/react-core';
+import { addNotification } from '@red-hat-insights/insights-frontend-components/components/Notifications';
+import ReactMarkdown from 'react-markdown/with-html';
+import * as AppActions from '../../AppActions';
+import { SYSTEM_TYPES, RISK_OF_CHANGE_LABEL, RISK_OF_CHANGE_DESC, TOTAL_RISK_LABEL, TOTAL_RISK_DESC } from '../../AppConstants';
```
Comment (AllenBW) [inline src/PresentationalComponents/RuleDetails/RuleDetails.js]: Checkout line eight, when your brining in a bunch of exports, might as well bring em all in under an alias

File: src/PresentationalComponents/RuleDetails/RuleDetails.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                                        <div>Total Risk</div>
+                                        <GridItem><Battery label={ TOTAL_RISK_LABEL[rule.total_risk] } severity={ rule.total_risk }/></GridItem>
+                                        <div>{ TOTAL_RISK_DESC[rule.total_risk] }</div>
+                                    </Grid>
+                                </GridItem>
+                                <GridItem sm={ 4 } md={ 12 }>
+                                    <div>Risk of Change</div>
+                                    <Shield
+                                        hasTooltip={ false }
+                                        impact={ RISK_OF_CHANGE_LABEL[rule.resolution_set.find(resolution => resolution.system_type === SYSTEM_TYPES.rhel).resolution_risk.risk] }
```
Comment (AllenBW) [inline src/PresentationalComponents/RuleDetails/RuleDetails.js]: D.R.Y. this out yo

File: src/PresentationalComponents/RuleDetails/RuleDetails.js (hunk 2 above)
Comment (bmclaughlin) [reply src/PresentationalComponents/RuleDetails/RuleDetails.js]: :+1:

File: src/PresentationalComponents/RuleDetails/RuleDetails.js (hunk 3 above)
Comment (bmclaughlin) [reply src/PresentationalComponents/RuleDetails/RuleDetails.js]: No doubt.

File: src/AppConstants.js (hunk 1 above)
Comment (bmclaughlin) [reply src/AppConstants.js]: Agreed, went this route to make progress as `var.1` was throwing an `Unexpected number` syuntax error.  Will be updating to an object and `var[1]` which doesn't complain.

**Comment by AllenBW** [comment]
Any update on breaking this up @bmclaughlin ? (the part yer having trouble with from the stuff thats done)

File: src/AppConstants.js [hunk 4]
```diff
@@ -59,3 +59,13 @@ export const FILTER_CATEGORIES = [
         ]
     }
 ];
+export const RISK_OF_CHANGE_DESC = {
+    1: 'The change takes very little time to implement and there is minimal impact to system operations. These are typically configuration changes that do not impact a running kernel or service. These changes can be made by modifying a line or file; the change that is taken is immediate and does not impact the system itself.', // eslint-disable-line max-len
```
Comment (AllenBW) [inline src/AppConstants.js]: `// eslint-disable-line max-len` should be done once, at the top of the file

File: package-lock.json [hunk 5]
```diff
@@ -114,7 +114,7 @@
         },
```
Comment (AllenBW) [inline package-lock.json]: No package.json changes, shouldn't be adding a lock file here

File: src/PresentationalComponents/RuleDetails/_RuleDetails.scss [hunk 6]
```diff
@@ -0,0 +1,7 @@
+.pf-l-grid.actions__detail .pf-l-grid__item.pf-u-pb-md  {
+    border-bottom: .01rem solid #ccc;
+}
+.ins-sev-clr-1 { color: var(--pf-global--Color--200); }
+.ins-sev-clr-2 { color: var(--pf-global--warning-color--100); }
+.ins-sev-clr-3 { color: #ec7a08; }
```
Comment (AllenBW) [inline src/PresentationalComponents/RuleDetails/_RuleDetails.scss]: no pf-global color for this? 😟

File: src/PresentationalComponents/RuleDetails/RuleDetails.js [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+                                            <span className='pf-u-pr-md' key={ key }>
+                                                { `${tagName}${rule.tags.split(' ').length !== (key + 1) ? ',' : ''}` }
+                                            </span>
+                                        )) }
+                                    </GridItem>
+                                ) }
+                            </Grid>
+                        </GridItem>
+                        <GridItem md={ 4 } sm={ 12 }>
+                            <Grid gutter='sm' className='actions__detail'>
```
Comment (AllenBW) [inline src/PresentationalComponents/RuleDetails/RuleDetails.js]: `className='actions__foo'` no longer needed/used

File: src/AppConstants.js (hunk 4 above)
Comment (bmclaughlin) [reply src/AppConstants.js]: :+1:

File: package-lock.json (hunk 5 above)
Comment (bmclaughlin) [reply package-lock.json]: Thought I had backed that out already, done!

File: src/PresentationalComponents/RuleDetails/_RuleDetails.scss (hunk 6 above)
Comment (bmclaughlin) [reply src/PresentationalComponents/RuleDetails/_RuleDetails.scss]: Unfortunately no pf-global, but I found a `--ins-color--orange` in ifc and am using that (same color).  Will update the Shield component to use it as well to ensure consistency.

File: src/PresentationalComponents/RuleDetails/RuleDetails.js [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+                                        <a href={ kbaDetails.view_uri }>
+                                            Knowledgebase Article
+                                        </a>
+                                    </GridItem>
+                                ) }
+                                { rule.tags && (
+                                    <GridItem>
+                                        Find Other Rules Related To:<br />
+                                        { rule.tags.split(' ').map((tagName, key) => (
+                                            <span className='pf-u-pr-md' key={ key }>
```
Comment (AllenBW) [inline src/PresentationalComponents/RuleDetails/RuleDetails.js]: Making each tag a span, each with its own `pf-u-pr-md` does not appear to jive with the mocks for this work (and is likely to get a lil wild when a rule has many tags), would recommend making it a string instead
screen shot 2019-03-06 at 12 59 55 pm

screen shot 2019-03-06 at 1 00 08 pm

File: src/PresentationalComponents/RuleDetails/RuleDetails.js [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+                                ) }
+                                <GridItem className='pf-u-pb-md' sm={ 8 } md={ 12 }>
+                                    <Grid className='ins-l-icon-group__vertical' sm={ 4 } md={ 12 }>
+                                        <div>Total Risk</div>
+                                        <GridItem>
+                                            <Battery
+                                                label=''
+                                                severity={ rule.total_risk }
+                                            />
+                                            <span className={ `label pf-u-pl-sm ins-sev-clr-${rule.total_risk}` }>
```
Comment (AllenBW) [inline src/PresentationalComponents/RuleDetails/RuleDetails.js]: This text should be centered  (vertically) on the battery.

File: src/PresentationalComponents/RuleDetails/RuleDetails.js [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+                                </GridItem>
+                                <GridItem sm={ 4 } md={ 12 } className='pf-t-light'>
+                                    <div>Risk of Change</div>
+                                    <Shield
+                                        hasTooltip={ false }
+                                        impact={ this.ruleResolutionRisk(rule) }
+                                        size={ 'md' }
+                                        title={ AppConstants.RISK_OF_CHANGE_LABEL[this.ruleResolutionRisk(rule)] }
+                                    />
+                                    <span className={ `label pf-u-pl-sm ins-sev-clr-${this.ruleResolutionRisk(rule)}` }>
```
Comment (AllenBW) [inline src/PresentationalComponents/RuleDetails/RuleDetails.js]: This text should be centered  (vertically) on the shield.

**Comment by bmclaughlin** [comment]
@AllenBW, knocked off those `span` tags and vertically centered the labels.

File: src/AppConstants.js [hunk 11]
```diff
@@ -59,3 +60,13 @@ export const FILTER_CATEGORIES = [
         ]
     }
 ];
+export const RISK_OF_CHANGE_DESC = {
+    1: 'The change takes very little time to implement and there is minimal impact to system operations. These are typically configuration changes that do not impact a running kernel or service. These changes can be made by modifying a line or file; the change that is taken is immediate and does not impact the system itself.',
+    2: 'Typically, these changes do not require that a system be taken offline. They are safe to make and even if executed incorrectly, will still leave the systemin a usable state.', // eslint-disable-line max-len
```
Comment (AllenBW) [inline src/AppConstants.js]: you can remove this and all of the other single line `max-len`s

File: src/PresentationalComponents/RuleDetails/RuleDetails.js [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+                                            />
+                                        </div>
+                                        <div className='pf-u-display-inline-flex'>
+                                            <span className={ `label pf-u-pl-sm ins-sev-clr-${rule.total_risk}` }>
+                                                &nbsp;{ AppConstants.TOTAL_RISK_LABEL[rule.total_risk] }
+                                            </span>
+                                        </div>
+                                    </div>
+                                    <p>
+                                        { `The likelihood that this will be a problem is ${AppConstants.LIKELIHOOD_LABEL[rule.likelihood]}.` }
```
Comment (AllenBW) [inline src/PresentationalComponents/RuleDetails/RuleDetails.js]: Probably want a space at the end of this period, renders on the page like so
screen shot 2019-03-07 at 6 53 09 am

File: src/PresentationalComponents/RuleDetails/RuleDetails.js [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+                                    <div className='pf-u-display-inline-flex pf-m-align-center pf-u-pb-sm pf-u-pt-sm'>
+                                        <div className='pf-u-display-inline-flex'>
+                                            <Battery
+                                                label=''
+                                                severity={ rule.total_risk }
+                                            />
+                                        </div>
+                                        <div className='pf-u-display-inline-flex'>
+                                            <span className={ `label pf-u-pl-sm ins-sev-clr-${rule.total_risk}` }>
+                                                &nbsp;{ AppConstants.TOTAL_RISK_LABEL[rule.total_risk] }
```
Comment (AllenBW) [inline src/PresentationalComponents/RuleDetails/RuleDetails.js]: So when this is undefined, the label goes away, maybe add an `|| 'Undefined' after `AppConstants.TOTAL_RISK_LABEL[rule.total_risk]` ?

**Pull request merged**